app/controllers/comments_controller.rb

```diff
@@ -16,21 +16,23 @@ def index
 
     if @podcast
       @user = @podcast
-      (@commentable = @user.podcast_episodes.find_by(slug: params[:slug])) || not_found
+      @commentable = @user.podcast_episodes.find_by(slug: params[:slug]) if @user.podcast_episodes
     else
       @user = User.find_by(username: params[:username]) ||
         Organization.find_by(slug: params[:username]) ||
         not_found
       @commentable = @root_comment&.commentable ||
-        @user.articles.find_by(slug: params[:slug]) ||
-        not_found
+        @user.articles.find_by(slug: params[:slug]) || nil
       @article = @commentable
-      not_found unless @commentable.published
+
+      not_found if @commentable && !@commentable.published
     end
 
-    @commentable_type = @commentable.class.name
+    @commentable_type = @commentable.class.name if @commentable
+
+    set_surrogate_key_header "comments-for-#{@commentable&.id}-#{@commentable_type}"
```

Comment:
If there is no commentable this header will just be "comments-for-" for every deleted comment. That seems like it could be a problem in Fastly, @rhymes or @benhalpern am I thinking about that right?

Reply:
Good call, I think we need something similar to what we're doing the API, thus building a surrogate key for the comment tree

Reply:
I had missed this. I not sure how you want to handle Fastly caching for this.

Reply:
Could we skip setting the header if commentable is not present? @benhalpern or @rhymes is that an option?

Reply:
It's a good idea. Rendering a single deleted comment shouldn't neither be heavy on computation nor frequent enough. Also it would be a cache with a single deleted comment, not that useful in the first place.

Reply:
I pushed up a commit to skip setting the header if commentable is not present.